test/SILOptimizer/concat_string_literals.swift

```diff
@@ -1,6 +1,8 @@
 // RUN: %target-swift-frontend -O -emit-ir  %s | %FileCheck %s
 // RUN: %target-swift-frontend -Osize -emit-ir  %s | %FileCheck %s
 
+// We have a different String representation on 32-bit architectures.
+// REQUIRES: CPU=arm64 || CPU=x86_64
```

Comment:
I think it's important to test some of this at least on 32-bit. A headache I've dealt with last week was finding just the right patterns so that StringObject can get constant folded for literals both large and small, and we need some regression test to check that.

Reply:
Sure! I added a variant of this test for 32-bit architectures.